---
description: 'Disallow unused private class members.'
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

> üõë This file is source code, not the primary documentation location! üõë
>
> See **https://typescript-eslint.io/rules/no-unused-private-class-members** for documentation.

This rule extends the base [`eslint/no-unused-private-class-members`](https://eslint.org/docs/rules/no-unused-private-class-members) rule.
It adds support for members declared with TypeScript's `private` keyword.

## Options

This rule has no options.

## Examples

<Tabs>
<TabItem value="‚ùå Incorrect">

```ts
class A {
  private foo = 123;
}
```

</TabItem>
<TabItem value="‚úÖ Correct">

```tsx
class A {
  private foo = 123;

  constructor() {
    console.log(this.foo);
  }
}
```

</TabItem>
</Tabs>

## Limitations

This rule does not detect the following cases:

(1) Nested classes using private members of the outer class:

```ts
class Foo {
  private prop = 123;

  method(a: Foo) {
    return class {
      // ‚úÖ Detected as a usage
      prop = this.prop;

      // ‚ùå NOT detected as a usage
      innerProp = a.prop;
    };
  }
}
```

(2) Usages of the private member outside of the class:

```ts
class Foo {
  private prop = 123;
}

const instance = new Foo();
// ‚ùå NOT detected as a usage
console.log(foo['prop']);
```

## When Not To Use It

If you don't want to be notified about unused private class members, you can safely turn this rule off.
